### PR TITLE
Relax consumer_callable requirements with a test

### PR DESCRIPTION
**Pull request by willstott101**
See #1577 - I just added and formatted the suggested test as the first of two commits.

> According to [ASGI specification](https://asgi.readthedocs.io/en/latest/specs/main.html#applications), both receive and send should be an awaitable callable, where as [create_task()](https://docs.python.org/3/library/asyncio-eventloop.html#creating-futures-and-tasks), accept only coroutine.
> 
> [NGINX Unit](http://unit.nginx.org/) ASGI server implementation, provides receive and send callable which returns Future, which in turns is awaitable and can be used just as a Task.
> 
> Using ensure_future() instead of create_task() is a simple way to create Task for coroutines and use Future if it is already provided.

### CODE REVIEW

**Comment by willstott101** [comment]
Test run here: https://github.com/willstott101/channels/actions/runs/1435192934

**Comment by willstott101** [comment]
I've edited the original comment here with more context from #1577 

This is a very minor non-breaking code change that would unblock usage of channels under **NGINX Unit**

Would the maintainers here prefer a more simplified unit test for `await_many_dispatch`? It seems to me that a unit test for a utility function is not something that would catch this if it regresses again later, after say, refactoring and removing that utility function. That being said I'm very happy to replace that test if it helps?

**Comment by willstott101** [comment]
Worth noting that literally less than 12 lines below this change [ensure_future is already used in this same function for later instantiations of callables](https://github.com/django/channels/blob/7bc96b6554a995471574ab498bfc942f79d099ad/channels/utils.py#L52) meaning this utility function isn't even consistent. The changes here both make the behaviour of this function consistent - and more spec compliant. The test ensures spec compliance in this area for the `AsyncHttpConsumer` in an integration test style.

Failing test run (with existing code): https://github.com/willstott101/channels/actions/runs/1909911872
Passing test run (with patched code): https://github.com/willstott101/channels/actions/runs/1909905681

**Review by carltongibson — APPROVED** [review]
Great. Thanks!

**Comment by ElRoberto538** [comment]
Is there any chance of this patch being back-ported to Channels 3? Updating to channels 4 is non-trivial for us, but without this patch we can't load our app in Nginx Unit.

**Comment by carltongibson** [comment]
@ElRoberto538 You'll have to back port it yourselves.

**Pull request merged**
